tollway/vehicle.py

```diff
@@ -0,0 +1,102 @@
+from random import choice
+from collections import deque
+
+import requests
+from bs4 import BeautifulSoup, Tag
+from faker import Faker
+from faker_vehicle import VehicleProvider
+from us.states import STATES_AND_TERRITORIES
+
+
+STATE_NAMES = [state.name for state in STATES_AND_TERRITORIES]
+TOLLWAYS_URL = "https://en.wikipedia.org/wiki/List_of_toll_roads_in_the_United_States"
+
+
+def create_vehicle(fake: Faker) -> dict:
+    
+    vehicle = {f"{key.lower()}": value for key, value in fake.vehicle_object().items()}
+    vehicle["license_plate"] = fake.license_plate()
+    vehicle["vin"] = fake.vin()
+    vehicle["state"] = choice(STATES_AND_TERRITORIES).abbr
+    vehicle["primary_color"] = fake.color_name()
+
+    return vehicle
+
+
+def _num_of_tables_between_states(soup: BeautifulSoup) -> dict:
+    states = {}
+    table_count_between_states = 0
+    state_name = ""
+    for row in soup.contents[0]:
+        if not isinstance(row, Tag):
+            continue
+
+        if isinstance(row, Tag) and row.name == "h2" and row.span.string in STATE_NAMES:
+            state_name = row.span.string
+            table_count_between_states = 0
+
+        if isinstance(row, Tag) and row.name == "table" and row["class"][0] == "wikitable" and state_name != "":
```

Comment:
you can simplify this with a helper function, also some duplication with the if above

Reply:
is the isinstance check needed? you already continued when it's not an instance of Tag